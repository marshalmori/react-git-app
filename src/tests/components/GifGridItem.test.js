import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Provas no <GifGridItem/>", () => {
  const title = "Um título";
  const url = "https://localhost/gif.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("deve mostrar o componente corretamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("deve ter um parágrafo com o título", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("deve ter uma imagem igual a url e alt dos props", () => {
    const img = wrapper.find("img");
    // console.log(img.html());
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("deve ter um animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
