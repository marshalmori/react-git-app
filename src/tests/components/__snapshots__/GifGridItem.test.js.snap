// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provas no <GifGridItem/> deve mostrar o componente corretamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Um título"
    src="https://localhost/gif.jpg"
  />
  <p>
    Um título
  </p>
</div>
`;
